examples/rest-grpc-multiplex/src/main.rs

```diff
@@ -32,19 +37,59 @@ impl Greeter for GrpcServiceImpl {
         tracing::info!("Got a request from {:?}", request.remote_addr());
 
         let reply = HelloReply {
-            message: format!("Hello {}!", request.into_inner().name),
+            message: format!("Hello {}, my name is {}.", request.into_inner().name, self.static_name),
         };
 
         Ok(TonicResponse::new(reply))
     }
 }
 
+/// axum Handler only takes one parameter (request),
+/// but tokio impls take two parameters (&self + request).
+/// tokio provides &self from an Arc stored in the service,
+/// so create GRPC_SERVICE as a static, pass it to tokio's GreeterServer::from_arc
+///   and statically reference it in the json_wrap_grpc.
+/// This ensures that all endpoints reference the same service.
+static GRPC_SERVICE: OnceCell<Arc<GrpcServiceImpl>> = OnceCell::new();
+
+/// Given a gRPC RPC implementation function,
+/// produce a closure that can be used as an axum Handler that:
+/// 1. deserializes JSON into the request type
+/// 2. calls the gRPC RPC implementation
+/// 3. serializes the response back to JSON
+fn json_wrap_grpc<'a, 'r, F, ReqT, ResT> (grpc_impl_func: F)
+    -> impl FnOnce(Json<ReqT>)
+        -> Pin<Box<dyn Future<Output = Result<Json<ResT>, GrpcErrorAsJson>> + Send + 'a>> + Clone + Send + Sized + 'static
+where
+    F: FnOnce(&'r GrpcServiceImpl, tonic::Request<ReqT>)
+        -> Pin<Box<dyn Future<Output = Result<tonic::Response<ResT>, tonic::Status>> + Send + 'r>> + Clone + Send + Sync + 'static,
+    for<'de> ReqT: serde::Deserialize<'de> + Send + 'a,
+    ResT: serde::Serialize
```

Comment:
I appreciate what you're trying to do but we cannot merge something this complicated as an example. That is just gonna confuse people. So unless you're able to find a simpler solution I don't think we can merge this.

Reply:
Yeah, I understand. Now that I've made the PR at least people who are looking to do this can find it from the discussion